src/inspector_agent.cc

```diff
@@ -586,9 +613,21 @@ bool Agent::StartIoThread(bool wait_for_connect) {
   }
 
   v8::Isolate* isolate = parent_env_->isolate();
+  HandleScope handle_scope(isolate);
+
+  // Enable tracking of async stack traces
+  if (!enable_async_hook_function_.IsEmpty()) {
+    Local<Function> enable_fn = enable_async_hook_function_.Get(isolate);
+    auto result = enable_fn->Call(parent_env_->context(),
+      Undefined(parent_env_->isolate()), 0, nullptr);
+    if (result.IsEmpty()) {
+      FatalError(
+          "node::InspectorAgent::StartIoThread",
+          "Cannot enable Inspector's AsyncHook, please report this.");
+    }
```

Comment:
There's been discussion about standardizing how `async_hooks`/`async_wrap` throws (cc @AndreasMadsen @addaleax), but the behavior in [Environment::AsyncHooks::pop_ids](https://github.com/nodejs/node/blob/v8.1.4/src/env-inl.h#L147-L159) and [fatalError() (`lib/async_hooks.js`)](https://github.com/nodejs/node/blob/v8.1.4/lib/async_hooks.js#L55-L68) is that we only abort on `--abort-on-uncaught-exception`, and otherwise we forcibly exit (without chance of recovery).

I'm down for discussing how this is handled, but consistency would be good. Possibly for this PR follow the other examples, and we can fix how it's handled in another PR.